src/equations/passive_tracers.jl

```diff
@@ -0,0 +1,222 @@
+end
+
+# Get the number of passive tracers
+@inline ntracers(::PassiveTracerEquations{NDIMS, NVARS, NTracers, FlowEquations}) where {NDIMS, NVARS,
+NTracers, FlowEquations} = NTracers
+
+function varnames(variables::typeof(cons2cons),
+                  tracer_equations::PassiveTracerEquations)
+    @unpack flow_equations = tracer_equations
+    flow_varnames = varnames(variables, flow_equations)
+    n_tracers = ntracers(tracer_equations)
+    return (flow_varnames..., ntuple(i -> "rho_xi_$i", Val(n_tracers))...)
+end
+
+function varnames(variables::typeof(cons2prim),
+                  tracer_equations::PassiveTracerEquations)
+    @unpack flow_equations = tracer_equations
+    flow_varnames = varnames(variables, flow_equations)
+    n_tracers = ntracers(tracer_equations)
+    return (flow_varnames..., ntuple(i -> "xi_$i", Val(n_tracers))...)
+end
+
+# Calculate flux for a single point
+@inline function flux(u, orientation_or_normal,
+                      tracer_equations::PassiveTracerEquations)
+    n_flow_variables = nvariables_flow(tracer_equations)
+
+    u_flow = flow_variables(u, tracer_equations)
+
+    v_normal = velocity(u_flow, orientation_or_normal, tracer_equations.flow_equations)
+
+    flux_tracer = SVector(ntuple(@inline(v->u[v + n_flow_variables] * v_normal),
+                                 Val(ntracers(tracer_equations))))
+
+    flux_flow = flux(u_flow, orientation_or_normal, tracer_equations.flow_equations)
+
+    return SVector(flux_flow..., flux_tracer...)
+end
+
+"""
+    initial_condition_density_wave(x, t, equations::PassiveTracerEquations)
+
+Takes the initial_condition_density_wave function for the flow equations and
+takes its translated first coordinates as the initial condition for the tracers.
+"""
+function initial_condition_density_wave(x, t,
+                                        equations::PassiveTracerEquations)
+    # Store translated coordinate for easy use of exact solution
+    u_flow = initial_condition_density_wave(x, t, equations.flow_equations)
+    # Obtain u_tracers by translating u_flow
+    xc = SVector(ntuple(_ -> 0.1f0 * one(eltype(x)), Val(ndims(equations))))
+
+    tracers = SVector((initial_condition_density_wave(x + i * xc, t,
+                                                      equations.flow_equations)[1] for i in 1:ntracers(equations))...)
+
+    u_tracers = u_flow[1] * tracers
+    return SVector(u_flow..., u_tracers...)
+end
+
+# Calculate the number of variables for the flow equations
+@inline function nvariables_flow(tracer_equations::PassiveTracerEquations)
+    return nvariables(tracer_equations.flow_equations)
+end
+
+# Obtain the flow variables from the conservative variables. The tracers are not included.
+@inline function flow_variables(u, tracer_equations::PassiveTracerEquations)
+    n_flow_variables = nvariables_flow(tracer_equations)
+    return SVector((u[i] for i in 1:n_flow_variables)...)
+end
```

Comment:
Is this type stable nowadays? Don't we need `ntuple` anymore? (the same pattern appears a few timers below)